scaling/provider/aws.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type AWSConfig struct {
+	GuestFramework *framework.Guest
+	HostFramework  *framework.Host
+	Logger         micrologger.Logger
+
+	ClusterID string
+}
+
+type AWS struct {
+	guestFramework *framework.Guest
+	hostFramework  *framework.Host
+	logger         micrologger.Logger
+
+	clusterID string
+}
+
+func NewAWS(config AWSConfig) (*AWS, error) {
+	if config.GuestFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestFramework must not be empty", config)
+	}
+	if config.HostFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostFramework must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+
+	a := &AWS{
+		guestFramework: config.GuestFramework,
+		hostFramework:  config.HostFramework,
+		logger:         config.Logger,
+
+		clusterID: config.ClusterID,
+	}
+
+	return a, nil
+}
+
+func (a *AWS) AddWorker() error {
+	customObject, err := a.hostFramework.AWSCluster(a.clusterID)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	patches := []framework.PatchSpec{
+		{
+			Op:    "add",
+			Path:  "/spec/aws/workers/-",
+			Value: customObject.Spec.AWS.Workers[0],
+		},
+	}
```

Comment:
All this is just copied from the `aws-operator` tests. The bits there got removed in favour of the implementation here.